packages/web3-providers-http/types/index.d.ts

```diff
@@ -46,7 +45,7 @@ export interface HttpProviderOptions {
 export class HttpProvider extends HttpProviderBase {
     host: string;
 
-    withCredentials: boolean;
+    withCredentials?: string;
```

Comment:
This will be breaking change, why we need it to be string?

Reply:
https://github.com/ChainSafe/web3.js/pull/5085#discussion_r884264478

Reply:
We can remain consistent with existing 1.x https://web3js.readthedocs.io/en/v1.7.4/web3.html#configuration and have `withCredentials` as boolean in web3.js options obj, and based on boolean value provided by user we can pass string to fetch ( include, omit ) [here](https://github.com/ChainSafe/web3.js/blob/535c9714549e494d257bd8832dd943f803efd558/packages/web3-providers-http/src/index.js#L113)  and it will go as [fetch wants](https://github.com/ChainSafe/web3.js/blob/535c9714549e494d257bd8832dd943f803efd558/packages/web3-providers-http/src/index.js#L159) plus we will not introduce breaking change.

Reply:
@jdevcs @nazarhussain Then it should be something more like `withCredentials?: boolean | string;` and handle boolean value with if block.

Reply:
@jdevcs  Please re-review.